applications/GeoMechanicsApplication/custom_processes/apply_k0_procedure_process.hpp

```diff
@@ -0,0 +1,161 @@
+
+// Project includes
+#include "includes/model_part.h"
+#include "utilities/openmp_utils.h"
+#include "utilities/math_utils.h"
+#include "includes/element.h"
+
+// Application includes
+
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class ApplyK0ProcedureProcess : public Process
+{
+  public:
+
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyK0ProcedureProcess);
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Constructor
+    ApplyK0ProcedureProcess(ModelPart&  model_part,
+                           Parameters& rParameters) : Process(Flags()), mrModelPart(model_part)
+    {
+        KRATOS_TRY
+
+        mModelPartName      =  rParameters["model_part_name"].GetString();
+
+        KRATOS_CATCH("")
+    }
+
+    ///------------------------------------------------------------------------------------
+
+    /// Destructor
+    ~ApplyK0ProcedureProcess() override{}
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    void ExecuteFinalizeSolutionStep() override
+    {
+        KRATOS_TRY
+
+        if (mrModelPart.NumberOfElements() > 0) {
+             // K0 procedure for the model part:
+             block_for_each(mrModelPart.Elements(), [this](Element& rElement) {
+                 CalculateK0Stresses(rElement);
+              });
+         }
+ 
+        KRATOS_CATCH("")
+    }
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+   
+  protected:
+    /// Member Variables
+
+    ModelPart& mrModelPart;
+    std::string mModelPartName;
+
+  private:
+
+      void CalculateK0Stresses(Element& rElement)
+      {
+          // Get K0 material parameters of this element ( probably there is something more efficient )
+          const Element::PropertiesType& rProp = rElement.GetProperties();
+          ConstitutiveLaw::Pointer pConstitutiveLaw = rProp.GetValue(CONSTITUTIVE_LAW);
+          const int& K0MainDirection = rProp[K0_MAIN_DIRECTION];
+          if (K0MainDirection < 0 || K0MainDirection > 1) {
+              KRATOS_ERROR << "undefined K0_MAIN_DIRECTION in ApplyK0ProcedureProcess: " << K0MainDirection << std::endl;
+          }
+          KRATOS_INFO("apply_k0_procedure_process") << " K0_NC " << rProp.Has(K0_NC) << " index voor Phi  " << rProp.Has(NUMBER_OF_UMAT_PHI_PARAMETER) << std::endl;
+
+          //Check for alternative K0 specifications
+          double PoissonUR = 0.;
+          if (rProp.Has(POISSON_UNLOADING_RELOADING)) PoissonUR = rProp[POISSON_UNLOADING_RELOADING];
```

Comment:
`PoissonUR` is declared and initialized here, but only used much later (at line 141). I'd rather have these two closer to each other. For instance, you could move it to just before the `for` loop at line 136.

Furthermore, you could combine the two statements using the ternary operator. The advantage in that case is that you can make `PoissonUR` immutable:
```suggestion
          const auto PoissonUR = rProp.Has(POISSON_UNLOADING_RELOADING) ? rProp[POISSON_UNLOADING_RELOADING] : 0.;
```

Reply:
Moved it closer, but still with the other material properties ( at 111 ).
Used the ternary operator.